goupnp.go

```diff
@@ -63,6 +64,9 @@ type MaybeRootDevice struct {
 	// the discovery of a device, regardless of if there was an error probing it.
 	Location *url.URL
 
+	// The address from which the device was discovered (if known - otherwise nil).
+	LocalAddr string
```

Comment:
We could keep this a string, but `net.IP` might be a better choice. This package knows that it will get a string from the `httpu` package containing an IP address, so it can call `net.ParseIP(...)` on that value and store a `net.IP` here and everywhere that follows from it.

That way we avoid user code needing to guess.

This also has the side effect of "otherwise nil" becoming accurate ;)